code/game/data_huds.dm

```diff
@@ -436,52 +436,47 @@
 /*~~~~~~~~~~~~
 	Circutry!
 ~~~~~~~~~~~~~*/
-/obj/item/electronic_assembly/proc/diag_hud_set_circuithealth(hide = FALSE)
+/atom/proc/diag_hud_set_circuithealth(percent, hide = FALSE)
```

Comment:
The component needs to handle all it's own functionality and interactions as much as possible except where that functionality would make more sense as a generic functionality of the object it's on. This is pretty clearly only for circuits so circuits should be handling it.